photon-core/src/test/java/org/photonvision/vision/pipeline/UncropApriltagTest.java

```diff
@@ -0,0 +1,108 @@
+                        TestUtils.getApriltagImagePath(TestUtils.ApriltagTestImages.kTag1_640_480, false),
+                        TestUtils.WPI2020Image.FOV,
+                        TestUtils.get2020LifeCamCoeffs(false));
+        frameProvider.requestFrameThresholdType(pipeline.getThresholdType());
+
+        CVPipelineResult pipelineResult;
+        pipelineResult = pipeline.run(frameProvider.get(), QuirkyCamera.DefaultCamera);
+
+        printTestResults(pipelineResult);
+
+        // Draw on input
+        var outputPipe = new OutputStreamPipeline();
+        var ret =
+                outputPipe.process(
+                        pipelineResult.inputAndOutputFrame, pipeline.getSettings(), pipelineResult.targets);
+
+        TestUtils.showImage(ret.inputAndOutputFrame.processedImage.getMat(), "Pipeline output", 999999);
+
+        pipeline.getSettings().static_x = 100;
+
+        CVPipelineResult croppedResults;
+        croppedResults = pipeline.run(frameProvider.get(), QuirkyCamera.DefaultCamera);
+
+        printTestResults(croppedResults);
+
+        var ret_cropped =
+                outputPipe.process(
+                        croppedResults.inputAndOutputFrame, pipeline.getSettings(), croppedResults.targets);
+
+        TestUtils.showImage(
+                ret_cropped.inputAndOutputFrame.processedImage.getMat(), "Cropped Pipeline output", 999999);
+        // these numbers are not *accurate*, but they are known and expected
+        var target = pipelineResult.targets.get(0);
+        var croppedTarget = croppedResults.targets.get(0);
+
+        // Test corner order
+        var corners = target.getTargetCorners();
+        var croppedCorners = croppedTarget.getTargetCorners();
+
+        Assertions.assertEquals(corners.get(0).x, croppedCorners.get(0).x, 0.1);
+        Assertions.assertEquals(corners.get(0).y, croppedCorners.get(0).y, 0.1);
+        Assertions.assertEquals(corners.get(1).x, croppedCorners.get(1).x, 0.1);
+        Assertions.assertEquals(corners.get(1).y, croppedCorners.get(1).y, 0.1);
+        Assertions.assertEquals(corners.get(2).x, croppedCorners.get(2).x, 0.1);
+        Assertions.assertEquals(corners.get(2).y, croppedCorners.get(2).y, 0.1);
+        Assertions.assertEquals(corners.get(3).x, croppedCorners.get(3).x, 0.1);
+        Assertions.assertEquals(corners.get(3).y, croppedCorners.get(3).y, 0.1);
+
+        var pose = target.getBestCameraToTarget3d();
+        var croppedPose = croppedTarget.getBestCameraToTarget3d();
+        // Test pose estimate translation and rotation
+        Assertions.assertEquals(pose.getTranslation().getX(), croppedPose.getTranslation().getX(), 0.2);
+        Assertions.assertEquals(pose.getTranslation().getY(), croppedPose.getTranslation().getY(), 0.2);
+        Assertions.assertEquals(pose.getTranslation().getZ(), croppedPose.getTranslation().getZ(), 0.2);
+        Assertions.assertEquals(pose.getRotation().getX(), croppedPose.getRotation().getX(), 0.2);
```

Comment:
0.2 meters/0.2 radians is a large bound for something we expect to be equal (or very close to it, thanks solvepnp iterative solver). Tighten this tolerance as much as possible.